lua/orgmode/org/mappings.lua

```diff
@@ -295,6 +295,20 @@ function OrgMappings:todo_prev_state()
   self:_todo_change_state('prev')
 end
 
+function OrgMappings:toggle_heading()
+  local node = Files.get_node_at_cursor()
+  local line = vim.fn.getline('.')
+  local parent_level = Files.get_closest_headline().level
+
+  if (node:type() == 'headline' or node:parent():type() == 'headline') then
+    line = line:gsub('^(%*+)', string.rep(' ', parent_level - 1))
```

Comment:
Make sure to check for `config.org_indent_mode` here. We do not need indent spaces if that option is not set to `indent`. You can find the similar check on several places in code.